lib/everest.cpp

```diff
@@ -851,19 +883,31 @@ void Everest::provide_cmd(const std::string& impl_id, const std::string& cmd_nam
             } catch (const std::exception& e) {
                 EVLOG_warning << fmt::format("Ignoring incoming cmd '{}' because not matching manifest schema: {}",
                                              cmd_name, e.what());
-                return;
+                error = CmdResultError{CmdEvent::SchemaValidationFailed, e.what()};
             }
         }
 
         // publish results
-        json res_data = json({});
-        res_data["id"] = data.at("id");
 
         // call real cmd handler
-        res_data["retval"] = handler(data.at("args"));
+        try {
+            if (not error.has_value()) {
+                res_data["retval"] = handler(data.at("args"));
+            }
+        } catch (const std::exception& e) {
+            EVLOG_error << fmt::format("Exception during handling of: {}->{}({}): {}",
+                                       this->config.printable_identifier(this->module_id, impl_id), cmd_name,
+                                       fmt::join(arg_names, ","), e.what());
+            error = CmdResultError{CmdEvent::HandlerException, e.what(), std::current_exception()};
+        } catch (...) {
+            EVLOG_error << fmt::format("Unknown exception during handling of: {}->{}({})",
+                                       this->config.printable_identifier(this->module_id, impl_id), cmd_name,
+                                       fmt::join(arg_names, ","));
+            error = CmdResultError{CmdEvent::HandlerException, "Unknown exception"};
```

Comment:
couldnt we use `std::current_exception()` here as well?

Reply:
added usage of `std::current_exception()`